vm/src/vm.rs

```diff
@@ -877,7 +877,7 @@ impl VirtualMachine {
     pub fn compile(
         &self,
         source: &str,
-        mode: &compile::Mode,
+        mode: compile::Mode,
```

Comment:
I like this change, but how did you get the idea to change this? Is this more efficient? Since the `Mode` is a simple enum, we can pass it by value easily, I agree.

Reply:
I've been thinking about it for a while, because as you said, there's really no reason to pass it by reference rather than value. It was also sorta annoying when passing them around to decide whether a function should take an `&Mode` or just a `Mode` and then pass a reference to that to `compile`.